sky/clouds/gcp.py

```diff
@@ -913,12 +912,31 @@ def _get_disk_type(cls,
                        disk_tier: Optional[resources_utils.DiskTier]) -> str:
         tier = cls._translate_disk_tier(disk_tier)
         tier2name = {
+            resources_utils.DiskTier.ULTRA: 'pd-extreme',
             resources_utils.DiskTier.HIGH: 'pd-ssd',
             resources_utils.DiskTier.MEDIUM: 'pd-balanced',
             resources_utils.DiskTier.LOW: 'pd-standard',
         }
+        if tier == resources_utils.DiskTier.ULTRA:
+            logger.warning(
+                'Using disk_tier=ultra on GCP will utilize extreme persistent disks, '
+                'which can lead to significant higher costs (~$2.2/h). '
+                'For more information, see: https://cloud.google.com/compute/disks-image-pricing#disk.'
+            )
         return tier2name[tier]
 
+    @classmethod
+    def _get_disk_specs(
+            cls,
+            disk_tier: Optional[resources_utils.DiskTier]) -> Dict[str, Any]:
+        return {
+            'disk_tier': cls._get_disk_type(disk_tier),
+            'disk_iops': 20000 if disk_tier == resources_utils.DiskTier.ULTRA
+                         else None,
+            # Only pd-extreme supports custom iops.
+            # see https://cloud.google.com/compute/docs/disks#disk-types
+        }
```

Comment:
nit: dont include an entry if not used
```suggestion
        specs = {'disk_tier': cls._get_disk_type(disk_tier)}
        if disk_tier == resources_utils.DiskTier.ULTRA:
            # Only pd-extreme supports custom iops.
            # see https://cloud.google.com/compute/docs/disks#disk-types
            specs['disk_iops'] = 20000
```